ext/stringio/stringio.c

```diff
@@ -1127,6 +1127,7 @@ prepare_getline_args(struct getline_arg *arg, int argc, VALUE *argv)
     long limit = -1;
 
     argc = rb_scan_args(argc, argv, "02:", &str, &lim, &opts);
+    int respect_chomp = argc == 0 || !NIL_P(str);
```

Comment:
Still supporting older ruby using C89.

```suggestion
    int respect_chomp;

    argc = rb_scan_args(argc, argv, "02:", &str, &lim, &opts);
    respect_chomp = argc == 0 || !NIL_P(str);
```